client/src/features/groupsV2/fields/EditGroupMemberModal.tsx

```diff
@@ -0,0 +1,173 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import cx from "classnames";
+import { useCallback, useEffect } from "react";
+import { PencilSquare, XLg } from "react-bootstrap-icons";
+import { Controller, useForm } from "react-hook-form";
+import {
+  Button,
+  Form,
+  Input,
+  Label,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+} from "reactstrap";
+
+import { RtkOrNotebooksError } from "../../../components/errors/RtkErrorAlert";
+import { Loader } from "../../../components/Loader";
+import type { GroupMemberResponse } from "../../projectsV2/api/namespace.api";
+import { usePatchGroupsByGroupSlugMembersMutation } from "../../projectsV2/api/projectV2.enhanced-api";
+import { ProjectMemberDisplay } from "../../projectsV2/shared/ProjectMemberDisplay";
+
+interface EditGroupMemberModalProps {
+  groupSlug: string;
+  isOpen: boolean;
+  member: GroupMemberResponse | undefined;
+  toggle: () => void;
+}
+
+export default function EditGroupMemberModal({
+  groupSlug,
+  isOpen,
+  member,
+  toggle,
+}: EditGroupMemberModalProps) {
+  return (
+    <Modal
+      backdrop="static"
+      centered
+      fullscreen="lg"
+      isOpen={isOpen && member != null}
+      size="lg"
+      toggle={toggle}
+    >
+      <ModalHeader toggle={toggle}>Change access</ModalHeader>
+      {member != null && (
+        <EditGroupMemberAccessForm
+          groupSlug={groupSlug}
+          member={member}
+          toggle={toggle}
+        />
+      )}
+    </Modal>
+  );
+}
+
+interface EditGroupMemberAccessFormProps {
+  groupSlug: string;
+  member: GroupMemberResponse;
+  toggle: () => void;
+}
+
+function EditGroupMemberAccessForm({
+  groupSlug,
+  member,
+  toggle,
+}: EditGroupMemberAccessFormProps) {
+  const [patchGroupMembers, { isLoading, isSuccess, error }] =
+    usePatchGroupsByGroupSlugMembersMutation();
+  const {
+    control,
+    formState: { isDirty },
+    handleSubmit,
+  } = useForm<GroupMemberResponse>({
+    defaultValues: {
+      id: member.id,
+      role: member.role,
+    },
+  });
+
+  useEffect(() => {
+    if (isSuccess) {
+      toggle();
+    }
+  }, [isSuccess, toggle]);
+
+  const onSubmit = useCallback(
+    (data: GroupMemberResponse) => {
+      patchGroupMembers({
+        groupSlug,
+        groupMemberPatchRequestList: [
+          {
+            id: data.id,
+            role: data.role,
+          },
+        ],
+      });
+    },
+    [groupSlug, patchGroupMembers]
+  );
+
+  return (
+    <>
+      <ModalBody className="pb-0">
```

Comment:
The bottom padding is removed here and added in the nested div. I'd suggest removing both

```suggestion
      <ModalBody>
```

Reply:
Question about these style adjustments: should these be updated on the `Project` components too? The modals where adapted from the `Project` case.

Reply:
Keeping them consistent would be great if you don't mind touching that code too.